frontend/src/container/GridCardLayout/GridCard/WidgetGraphComponent.tsx

```diff
@@ -131,15 +131,22 @@ function WidgetGraphComponent({
 
 		const uuid = v4();
 
+		// this is added to make sure the cloned panel is of the same dimensions as the original one
+		const originalPanelLayout = selectedDashboard.data.layout?.find(
+			(l) => l.i === widget.id,
```

Comment:
check if we need any memoisation here.

Reply:
memoization inside a user-triggered action? this will be called on click of `clonehandler` and should always fetch the latest results during that time